src/main/java/com/hubspot/jinjava/lib/fn/MacroFunction.java

```diff
@@ -140,11 +140,15 @@ public String getEvaluationResult(
     LengthLimitingStringBuilder result = new LengthLimitingStringBuilder(
       interpreter.getConfig().getMaxOutputSize()
     );
+    int initialEagerTokenCount = interpreter.getContext().getEagerTokens().size();
 
     for (Node node : content) {
       result.append(node.render(interpreter));
     }
-    if (interpreter.getConfig().getExecutionMode().isPreserveRawTags()) {
+    if (
+      interpreter.getConfig().getExecutionMode().isPreserveRawTags() &&
+      initialEagerTokenCount == interpreter.getContext().getEagerTokens().size()
+    ) {
```

Comment:
Unrelated, but saw in my testing, don't need to unwrap the raw tags from a macro function if it has deferred tokens within it.